tasks/EchoesToFrameConverterTask.cpp

```diff
@@ -0,0 +1,172 @@
+
+#include "EchoesToFrameConverterTask.hpp"
+#include <base-logging/Logging.hpp>
+#include <base/samples/RigidBodyState.hpp>
+#include <math.h>
+#include <opencv2/imgproc/imgproc.hpp>
+
+using namespace radar_base;
+using namespace cv;
+using namespace base::samples::frame;
+EchoesToFrameConverterTask::EchoesToFrameConverterTask(std::string const& name)
+    : EchoesToFrameConverterTaskBase(name)
+{
+}
+
+EchoesToFrameConverterTask::~EchoesToFrameConverterTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See EchoesToFrameConverterTask.hpp for more detailed
+// documentation about them.
+
+bool EchoesToFrameConverterTask::configureHook()
+{
+    if (!EchoesToFrameConverterTaskBase::configureHook())
+        return false;
+    configureOutput();
+    updateLookUpTable();
+    return true;
+}
+bool EchoesToFrameConverterTask::startHook()
+{
+    if (!EchoesToFrameConverterTaskBase::startHook())
+        return false;
+    m_last_sample = base::Time::now();
+    return true;
+}
+void EchoesToFrameConverterTask::updateHook()
+{
+    EchoesToFrameConverterTaskBase::updateHook();
+
+    double yaw_correction = 0;
+    base::samples::RigidBodyState sensor2ref_pose;
+    if (_sensor2ref_pose.read(sensor2ref_pose) != RTT::NoData) {
+        yaw_correction = sensor2ref_pose.getYaw();
+    }
+
+
+    Radar radar_echo;
+    while (_echo.read(radar_echo, false) == RTT::NewData) {
+        if (m_current_sweep_size != radar_echo.sweep_length ||
+            m_current_num_angles != (2 * M_PI / radar_echo.step_angle.getRad()) ||
+            m_current_range != radar_echo.range) {
+            m_current_sweep_size = radar_echo.sweep_length;
+            m_current_num_angles = 2 * M_PI / radar_echo.step_angle.getRad();
+            m_current_range = radar_echo.range;
+            updateLookUpTable();
+            m_echoes.clear();
+        }
+
+        addEchoesToFrame(radar_echo, yaw_correction);
+    }
+    if (base::Time::now() - m_last_sample > _export_config.get().time_between_frames) {
+        publishFrame();
+    }
+}
+void EchoesToFrameConverterTask::errorHook()
+{
+    EchoesToFrameConverterTaskBase::errorHook();
+}
+void EchoesToFrameConverterTask::stopHook()
+{
+    EchoesToFrameConverterTaskBase::stopHook();
+}
+void EchoesToFrameConverterTask::cleanupHook()
+{
+    EchoesToFrameConverterTaskBase::cleanupHook();
+}
+
+void EchoesToFrameConverterTask::updateLookUpTable()
+{
+    LOG_INFO_S << "Updating LookUpTable";
+    auto config = _export_config.get();
+    m_lut.reset(new EchoToImageLUT(m_current_num_angles,
+        m_current_sweep_size,
+        config.beam_width,
+        config.window_size));
+}
+
+void EchoesToFrameConverterTask::addEchoesToFrame(Radar const& echo,
+    double yaw_correction)
+{
+    LOG_INFO_S << "Adding Echoes to frame";
+    double angle = echo.start_heading.getRad() + yaw_correction;
```

Comment:
This can be over 2*PI because of the yaw_correction. You should be using base::Angle for yaw_correction too, which would solve the problem.